packages/zoe/src/contractSupport/zoeHelpers.js

```diff
@@ -72,13 +73,23 @@ export const swap = (zcf, leftSeat, rightSeat) => {
 /** @type {SwapExact} */
 export const swapExact = (zcf, leftSeat, rightSeat) => {
   try {
-    rightSeat.decrementBy(harden(rightSeat.getProposal().give));
-    leftSeat.incrementBy(harden(leftSeat.getProposal().want));
-
-    leftSeat.decrementBy(harden(leftSeat.getProposal().give));
-    rightSeat.incrementBy(harden(rightSeat.getProposal().want));
-
-    zcf.reallocate(leftSeat, rightSeat);
+    atomicTransfer(
+      zcf,
+      harden([
+        [
+          rightSeat,
+          leftSeat,
+          rightSeat.getProposal().give,
+          leftSeat.getProposal().want,
+        ],
+        [
+          leftSeat,
+          rightSeat,
+          leftSeat.getProposal().give,
+          rightSeat.getProposal().want,
```

Comment:
ditto

Reply:
Yes! This one needed to split up for the reason that I initially missed. The old TransferArgs lines before were not individually balanced.

Done.